Include external reporting template for parsed results too

marmarek opened the pull request:
Allow branding to set external reporting link to parsed results too.
Since parsed results are all rendered as part of a single template,
module and step id needs to be set directly in a template during
the iteration. And branding template modified to actually use those
values.

I'm not an expert in CSS, and this is the best I can do - the action link is at the bottom right of expanded parsed result text. If anyone know how to put it at the top right (but not increasing height of the collapsed row), that would be more consistent with other places.
You can see how it looks now here: https://openqa.qubes-os.org/tests/4710#step/QubeManagerTest/69

I've included changes to openSUSE branding, but in fact I've tested it on a Qubes OS one (not published yet). BTW is there a way to load branding from an alternative directory?

okurz commented on templates/test/module_table.html.ep:
```diff
@@ -49,7 +49,11 @@
                             % if ($is_parser_text_result) {
                                 <span title="<%= $title %>" data-href="<%= $href %>" onclick="toggleTextPreview(this)" class="text-result">
-                                    <span class="resborder <%= $resborder %>"><%= $step->{text_data} %></span>
+                                    <span class="resborder <%= $resborder %>"><%= $step->{text_data} %><span class="step_actions" style="float: right; white-space: normal;">
```
please but the inner `<span>` into a new indented line, the closing `</span>` accordingly. It is ok to test this style changes here but to have it merge-able I recommend to put that value for style in an according CSS style file. You can probably experiment with this more easily within your browser developer tools. Finally this should probably go into assets/stylesheets/test-details.scss . The look is ok for me though.

@Martchus do you have an idea to answer the question regarding putting the icon top-right w/o increasing the row size?

marmarek replied:
> please but the inner `<span>` into a new indented line, the closing `</span>` accordingly.

I've tried, but since this span has `white-space: pre-wrap`, it will cause that newline to actually affect the visual effect. That's why `<%= $step->{text_data} %>` is on the same line here too.

Martchus replied:
I needed to checkout myself. But `style="float: right; white-space: normal;` should better go into its own CSS class.

> I've tried, but since this span has white-space: pre-wrap

Yes, without changing the HTML/CSS more (e.g. having only the `text_data` text within its own span) it is not possible to change this. I don't think it is important, though.

marmarek commented:
> The diff coverage is `0%`.

That's weird, as my initial commit (see force-push history) definitely caused some test failures, so it was executed in some of the tests. Is test coverage measured only on a subset of the tests?

okurz commented:
the external branding is certainly not covered by any coverage tests but your other changes should be covered. I wouldn't worry about it though. I could not find any test failures in your previous commit but only the same coverage missing check that failed there.

marmarek commented:
The previous failure is here: https://circleci.com/gh/os-autoinst/openQA/6066

Martchus commented on templates/test/module_table.html.ep:
```diff
@@ -49,7 +49,11 @@
                                 <span title="<%= $title %>" data-href="<%= $href %>" onclick="toggleTextPreview(this)" class="text-result">
-                                    <span class="resborder <%= $resborder %>"><%= $step->{text_data} %></span>
+                                    <span class="resborder <%= $resborder %>"><%= $step->{text_data} %><span class="step_actions" style="float: right; white-space: normal;">
+                                        % stash('moduleid', $module->{name});
```
Stashing things within the template code looks wrong. It would make more sense to pass these variables as arguments to the `bug_report_actions` helper and then pass them to `render_to_string`.

marmarek commented:
Done the changes:
 - moved style to css file (actually didn't need to introduce new class, it was enough to set it for `step_actions` nested in `resborder`)
 - added parameters to `bug_report_actions`, instead of stashing thing inside template

The second one I've done in a generic way (any arguments as passed down to the template). If you prefer to explicitly name just module and step id, let me know.

Martchus commented:
@marmarek Thanks for changing it according to my suggestions. I've also added a commit to tweak the markup/styling a little bit. I guess that's more efficient than explaining it (see the commit message for details).

okurz commented:
I reported https://progress.opensuse.org/issues/60035 for the unrelated OBS check failure in plugin_obs_rsync_async
